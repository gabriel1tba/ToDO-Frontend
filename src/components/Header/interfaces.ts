export interface IHeader {
  userName?: string;
  onSignOut: () => void;
}
