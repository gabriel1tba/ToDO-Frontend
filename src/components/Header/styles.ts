import styled from 'styled-components';

export const Wrapper = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100vw;
  height: 90px;
  padding: 0 15rem;

  box-shadow: 0px 0px 3px rgba(0, 0, 0, 0.4);

  img {
    transform: scale(0.8);
  }

  > div {
    margin-right: 450px;

    > h3 {
      color: #ff9000;
    }
  }
  > button {
    background: none;
    border: none;
  }
`;
