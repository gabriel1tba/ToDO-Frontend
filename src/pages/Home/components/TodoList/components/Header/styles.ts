import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;

  gap: 0.5rem;

  input {
    height: 54px;
    width: 100%;
    padding: 1rem;

    border: 1px solid #0d0d0d;
    border-radius: 8px;

    font-size: 1.5rem;
  }

  button {
    font-size: 1rem;
    height: 54px;
    width: 160px;
    font-weight: 700;

    svg {
      font-size: 1rem;
    }
  }

  @media (max-width: 576px) {
    flex-direction: column;

    button {
      min-width: 100%;
      margin-top: 0.5rem;
    }
  }
`;
