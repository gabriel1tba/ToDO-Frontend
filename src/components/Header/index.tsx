import { RiShutDownLine } from 'react-icons/ri';

import { useAuth } from 'hooks';

import logoImg from 'assets/logo.png';

import * as S from './styles';

export interface IHeader {
  userName?: string;
  onSignOut: () => void;
}

const Header = ({ userName }: IHeader) => {
  const { signOut } = useAuth();

  return (
    <S.Wrapper>
      <S.HeaderContent>
        <div>
          <img src={logoImg} alt="logo image" />

          <div>
            {userName ? (
              <>
                <h4>Seja bem-vindo,</h4>
                <h3>{userName}</h3>
              </>
            ) : null}
          </div>
        </div>
        <div>
          <button onClick={signOut}>
            <RiShutDownLine size={40} />
          </button>
        </div>
      </S.HeaderContent>
    </S.Wrapper>
  );
};

export default Header;
