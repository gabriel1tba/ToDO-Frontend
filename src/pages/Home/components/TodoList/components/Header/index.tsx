import { BiPlusCircle } from 'react-icons/bi';

import Button from 'components/Button';

import * as S from './styles';

interface IHeaderProps {
  searchTerm: string;
  onSearchTerm: (value: string) => void;
  onToggleModal: () => void;
}

const Header = ({ searchTerm, onSearchTerm, onToggleModal }: IHeaderProps) => (
  <S.Wrapper>
    <input
      value={searchTerm}
      onChange={(e) => onSearchTerm(e.target.value)}
      placeholder="Buscar..."
    />

    <Button variant="primary" icon={<BiPlusCircle />} onClick={onToggleModal}>
      Nova tarefa
    </Button>
  </S.Wrapper>
);

export default Header;
