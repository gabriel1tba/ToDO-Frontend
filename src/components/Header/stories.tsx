import { Story, Meta } from '@storybook/react';

import Header from '.';
import { IHeader } from './interfaces';

export default {
  title: 'Header',
  component: Header,
  argTypes: {
    userName: {
      type: 'string',
    },
    onSignOut: {
      type: 'function',
    },
  },
} as Meta;

export const Default: Story<IHeader> = (args) => <Header {...args} />;

Default.args = {
  userName: 'Gabriel Ferreira',
  onSignOut: () => null,
};
