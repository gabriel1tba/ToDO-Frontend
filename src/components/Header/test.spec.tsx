import { render } from '@testing-library/react';

import Header from '.';

describe('<Header />', () => {
  it('should render the Header correctly', () => {
    render(<Header userName="Gabriel" onSignOut={() => ({})} />);

    // expect(container.firstChild).toMatchSnapshot();
  });
});
