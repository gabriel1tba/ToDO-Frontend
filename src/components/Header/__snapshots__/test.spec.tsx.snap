// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header correctly 1`] = `
.c0 {
  width: 100vw;
  height: 120px;
  border-bottom: 1px solid rgba(0,0,0,0.1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1032px;
  padding: 0 2rem;
}

.c1 div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 div:first-child div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 div:first-child div h3 {
  color: #ff9000;
}

.c1 div:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 div:last-child > button {
  background: none;
  border: none;
}

.c1 div:last-child > button svg {
  color: #ff9000;
}

@media (max-width:540px) {
  .c1 div:first-child div h3,
  .c1 div:first-child div h4 {
    display: none;
  }
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <img
        alt="logo image"
        src="logo.png"
      />
      <div>
        <h4>
          Seja bem-vindo,
        </h4>
        <h3>
          Gabriel Ferreira
        </h3>
      </div>
    </div>
    <div>
      <button>
        <svg
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6.265 3.807l1.147 1.639a8 8 0 1 0 9.176 0l1.147-1.639A9.988 9.988 0 0 1 22 12c0 5.523-4.477 10-10 10S2 17.523 2 12a9.988 9.988 0 0 1 4.265-8.193zM11 12V2h2v10h-2z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</header>
`;
